src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/Repository/ReleaseVersionRepositoryTests.cs

```diff
@@ -22,186 +21,95 @@ public class GetLatestPublishedReleaseVersionTests : ReleaseVersionRepositoryTes
-                .Single(rv => rv is { Published: not null, Year: 2021, Version: 1 });
-
-            Assert.NotNull(result);
-            Assert.Equal(expectedReleaseVersion.Id, result.Id);
-        }
-
-        [Fact]
-        public async Task PublicationHasNoPublishedReleaseVersions_ReturnsNull()
-        {
-            var publications = _dataFixture
-                .DefaultPublication()
-                // Index 0 has an unpublished release version
-                // Index 1 has a published release version
-                .ForIndex(0, p => p.SetReleases(_dataFixture
-                    .DefaultRelease(publishedVersions: 0, draftVersion: true)
-                    .Generate(1)))
-                .ForIndex(1, p => p.SetReleases(_dataFixture
-                    .DefaultRelease(publishedVersions: 1)
-                    .Generate(1)))
-                .GenerateList(2);
-
-            var contextId = await AddTestData(publications);
-            await using var contentDbContext = InMemoryContentDbContext(contextId);
-            var repository = BuildRepository(contentDbContext);
-
-            Assert.Null(await repository.GetLatestPublishedReleaseVersion(publications[0].Id));
-        }
-
-        [Fact]
-        public async Task PublicationHasNoReleaseVersions_ReturnsNull()
-        {
-            var publications = _dataFixture
-                .DefaultPublication()
-                // Index 0 has no release versions
-                // Index 1 has a published release version
-                .ForIndex(1, p => p.SetReleases(_dataFixture
-                    .DefaultRelease(publishedVersions: 1)
-                    .Generate(1)))
-                .GenerateList(2);
-
-            var contextId = await AddTestData(publications);
-            await using var contentDbContext = InMemoryContentDbContext(contextId);
-            var repository = BuildRepository(contentDbContext);
-
-            Assert.Null(await repository.GetLatestPublishedReleaseVersion(publications[0].Id));
-        }
-
-        [Fact]
-        public async Task PublicationDoesNotExist_ReturnsNull()
-        {
-            Publication publication = _dataFixture
-                .DefaultPublication()
-                .WithReleases(_dataFixture
-                    .DefaultRelease(publishedVersions: 1)
-                    .Generate(1));
-
-            var contextId = await AddTestData(publication);
-            await using var contentDbContext = InMemoryContentDbContext(contextId);
-            var repository = BuildRepository(contentDbContext);
-
-            Assert.Null(await repository.GetLatestPublishedReleaseVersion(Guid.NewGuid()));
-        }
+                .WithReleases(_ =>
+                [
+                    _dataFixture.DefaultRelease(publishedVersions: 1, year: 2022),
+                    _dataFixture.DefaultRelease(publishedVersions: 2, draftVersion: true, year: 2021)
+                ])
+                .GenerateTuple2();
```

Comment:
Wonder if we should do the same thing as `GetLatestReleaseVersionTests.Success` here, and set the `ReleaseSeries` explicitly. Purposefully make `2021` go first in the series, but give it **zero** published versions. And that way we can check that the `2021` release version should not be the one fetched back.

Reply:
It's not super obvious, but the only overload of `GetLatestPublishedReleaseVersion` which remains now is the variant with parameters `Guid publicationId, string releaseSlug`. 

There can only ever be one release in the publication with the specified slug, so the release series order is not relevant. It's still returned even if it's not the latest release for the publication, because the method is used to retrieve the latest version of a specific release. Up until the work that @rianski is doing in EES-5219 this would usually have been a non-latest release.

We also have a  test `PublicationHasNoPublishedReleaseVersions_ReturnsNull` where `2021` is given zero published versions and checks there's no result.

The overload with just  `Guid publicationId` was removed in favour of replacing all occurrences to access it directly from field `Publication.LatestPublishedReleaseVersion` instead of having a method which calculated it based on the release series order and working out the latest version for the latest release.

To make things a bit clearer I've renamed the method from `GetLatestPublishedReleaseVersion` to `GetLatestPublishedReleaseVersionByReleaseSlug` in a new commit, and renamed the inner test class too.

I also change the implementation of that method to use `SingleOrDefaultAsync` rather than `FirstOrDefaultAsync`.

I think your comment might be valid for other methods though and this has reminded me that we should be checking the order of results so I'm going to review the other tests.

Reply:
Changes look good! :)